clients/java/client/src/main/java/org/operaton/bpm/client/impl/ExternalTaskClientBuilderImpl.java

```diff
@@ -246,7 +252,9 @@ public ExternalTaskClient build() {
   }
 
   protected void initBaseUrl() {
-    baseUrl = sanitizeUrl(baseUrl);
+    if (this.urlResolver instanceof PermanentUrlResolver) {
```

Comment:
Replace this instanceof check and cast with 'instanceof PermanentUrlResolver permanentUrlResolver'